packages/server/src/middleware/authorized.js

```diff
@@ -5,8 +5,33 @@ const {
   BUILDER_LEVEL_ID,
   BUILDER,
 } = require("../utilities/accessLevels")
+const environment = require("../environment")
+const { apiKeyTable } = require("../db/dynamoClient")
 
 module.exports = (permName, getItemId) => async (ctx, next) => {
+  if (
+    environment.CLOUD &&
+    ctx.headers["x-api-key"] &&
+    ctx.headers["x-instanceid"]
+  ) {
+    // api key header passed by external webhook
+    const apiKeyInfo = await apiKeyTable.get({
+      primary: ctx.headers["x-api-key"],
+    })
+
+    if (apiKeyInfo) {
+      ctx.isAuthenticated = true
```

Comment:
should we consider putting all of these flags under an `auth` object? Just for the sake of knowing what to look for, all the flags can be set in an object rather than needing to search the code for all the various `is...` booleans set in the `ctx`?

Reply:
Yep, I think that's a pretty good idea. It's getting a bit tedious to have to set all these separate things. I'll update that across the board 👍